crates/hyperswitch_domain_models/src/payments.rs

```diff
@@ -82,70 +82,179 @@ impl PaymentIntent {
+    tax_details: Option<TaxDetails>,
+    /// The action to whether calculate tax by calling external tax provider or not
+    skip_external_tax_calculation: TaxCalculationOverride,
+    /// The action to whether calculate surcharge or not
+    skip_surcharge_calculation: SurchargeCalculationOverride,
+    /// The surcharge amount to be added to the order, collected from the merchant
+    surcharge_amount: Option<MinorUnit>,
+    /// tax on surcharge amount
+    tax_on_surcharge: Option<MinorUnit>,
+}
+
+#[cfg(feature = "v2")]
+impl AmountDetails {
+    /// Get the action to whether calculate surcharge or not as a boolean value
+    fn get_surcharge_action_as_bool(&self) -> bool {
+        match self.skip_surcharge_calculation {
+            SurchargeCalculationOverride::Skip => false,
+            SurchargeCalculationOverride::Calculate => true,
+        }
+    }
+
+    /// Get the action to whether calculate external tax or not as a boolean value
+    fn get_external_tax_action_as_bool(&self) -> bool {
+        match self.skip_external_tax_calculation {
+            TaxCalculationOverride::Skip => false,
+            TaxCalculationOverride::Calculate => true,
+        }
+    }
+}
+
 #[cfg(all(feature = "v2", feature = "payment_v2"))]
 #[derive(Clone, Debug, PartialEq, serde::Serialize)]
 pub struct PaymentIntent {
+    /// The global identifier for the payment intent. This is generated by the system.
+    /// The format of the global id is `{cell_id:5}_pay_{time_ordered_uuid:32}`.
+    pub id: id_type::PaymentGlobalId,
+    /// The identifier for the merchant. This is automatically derived from the api key used to create the payment.
     pub merchant_id: id_type::MerchantId,
+    /// The status of payment intent.
     pub status: storage_enums::IntentStatus,
-    pub amount: MinorUnit,
-    pub currency: Option<storage_enums::Currency>,
+    /// The amount related details of the payment
+    pub amount_details: AmountDetails,
+    /// The total amount captured for the order. This is the sum of all the captured amounts for the order.
     pub amount_captured: Option<MinorUnit>,
+    /// The identifier for the customer. This is the identifier for the customer in the merchant's system.
     pub customer_id: Option<id_type::CustomerId>,
+    /// The description of the order. This will be passed to connectors which support description.
     pub description: Option<String>,
+    /// The return url for the payment. This is the url to which the user will be redirected after the payment is completed.
     pub return_url: Option<String>,
-    pub metadata: Option<serde_json::Value>,
-    pub statement_descriptor_name: Option<String>,
+    /// The metadata for the payment intent. This is the metadata that will be passed to the connectors.
+    pub metadata: Option<pii::SecretSerdeValue>,
+    /// The statement descriptor for the order, this will be displayed in the user's bank statement.
+    pub statement_descriptor: Option<String>,
```

Comment:
This can be a `Option<StatementDescriptor>`, type can have a max length validation on new.